contrib/babelfishpg_common/src/collation.c

```diff
@@ -1205,20 +1238,20 @@ BABELFISH_CLUSTER_COLLATION_OID()
 {
 	if (sql_dialect == SQL_DIALECT_TSQL)
 	{
-		get_server_collation_oid_internal(false);	/* set and cache
+		Oid db_coll = get_database_or_server_collation_oid_internal(false);	/* set and cache
 													 * server_collation_oid */
 
-		if (OidIsValid(server_collation_oid))
-			return server_collation_oid;
+		if (OidIsValid(db_coll))
+			return db_coll;
 	}
 	return DEFAULT_COLLATION_OID;
 }
 
 /*
- * collation_is_CI_AS - Returns true if collation with given colloid is CI_AS.
+ * collation_is_CI - Returns true if collation with given colloid is CI.
  */
 bool
-collation_is_CI_AS(Oid colloid)
+collation_is_CI(Oid colloid)
```

Comment:
This function is used in some places to specifically check for CI_AS.
If we want to check for CI collations only, we can create a new helper function

Reply:
All the functions which uses this function ideally searches for either non-deterministic or CI collation...so no need to introduce anything new

Reply:
```
CREATE OR REPLACE FUNCTION sys.replace (input_string sys.VARCHAR, pattern sys.VARCHAR, replacement sys.VARCHAR)
RETURNS sys.VARCHAR AS
$BODY$
BEGIN
   if PG_CATALOG.length(pattern) = 0 then
       return input_string;
   elsif sys.is_collated_ai(input_string) then
       return pg_catalog.replace(input_string, pattern, replacement);
   elsif sys.is_collated_ci_as(input_string) then
       return regexp_replace(input_string, '***=' || pattern, replacement, 'ig');
   else
       return regexp_replace(input_string, '***=' || pattern, replacement, 'g');
   end if;
END
$BODY$
LANGUAGE plpgsql IMMUTABLE PARALLEL SAFE STRICT;
```
sys.replace relies on it to check for CI_AS

Reply:
Seems like we need to update our definition of sys.replace because
first, Following condition would handle `[CI | CS]_AI` collation 
```
elsif sys.is_collated_ai(input_string) then return pg_catalog.replace(input_string, pattern, replacement);
```
so in second condition, we only need to check `CI` and it would handle `CI_AS` part automatically. 

But look for other references also.

Reply:
Please check, 

1. `transform_like_in_add_constraint`